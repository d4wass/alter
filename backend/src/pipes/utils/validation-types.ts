import { ValidationError } from '@nestjs/common';

export type ConvertedValidationError = {
  property: string;
  constraints: any;
  children?: ValidationError[];
};

export type ConstructorTypes =
  | StringConstructor
  | BooleanConstructor
  | NumberConstructor
  | ArrayConstructor
  | ObjectConstructor;
