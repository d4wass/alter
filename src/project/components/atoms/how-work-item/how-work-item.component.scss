.wrapper {
    display: flex;
}


.indicator {
    margin-right: 20px;
    p{
        border-radius: 50%;
        border: solid 1px #000;
        padding: 10px 15px;
    }
}
.content {
    display: flex;
    flex-direction: column;
}