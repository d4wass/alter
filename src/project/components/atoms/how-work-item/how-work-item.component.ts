import { Component, Input } from '@angular/core';

@Component({
  selector: 'app-how-work-item',
  template: `
    <div class="wrapper">
      <div class="indicator">
        <p *ngIf="itemNumber" data-test="indicator">{{ itemNumber }}</p>
      </div>
      <div class="content">
        <h2>{{ itemTitle }}</h2>
        <p>{{ itemContent }}</p>
      </div>
    </div>
  `,
  styleUrls: ['./how-work-item.component.scss']
})
export class HowWorkItemComponent {
  @Input() itemNumber?: string;
  @Input() itemTitle!: string;
  @Input() itemContent!: string;
}
