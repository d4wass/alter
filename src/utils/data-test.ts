export function dataTest(selector: string) {
  return `[data-test="${selector}"]`;
}
