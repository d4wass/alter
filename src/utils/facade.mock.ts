import { Observable, BehaviorSubject } from 'rxjs';

export type FacadeMock<T> = {
  [P in keyof T]: T[P] extends Observable<infer U>
    ? BehaviorSubject<U>
    : T[P] extends (args: infer A) => Observable<infer R>
    ? (args: A) => BehaviorSubject<R>
    : T[P];
};
