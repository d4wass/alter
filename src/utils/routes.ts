export interface urlItem {
  description: string;
  url: string;
}

export const urlList: urlItem[] = [
  { description: 'Learn more', url: 'faq' },
  { description: 'Carculator', url: 'carculator' },
  { description: 'Become a host', url: 'host' },
  { description: 'Create Account', url: 'new-user' },
  { description: 'Your Cars', url: 'vehicles' }
];
