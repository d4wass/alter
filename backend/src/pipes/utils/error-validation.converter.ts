import { ValidationError } from 'class-validator';
import { ConvertedValidationError } from './validation-types';

export class ValidationErrorConverter {
  private _convertedErrors: Array<ConvertedValidationError> = [];
  public get convertedErrors(): Array<ConvertedValidationError> {
    return this._convertedErrors;
  }
  public set convertedErrors(value: Array<ConvertedValidationError>) {
    this._convertedErrors = value;
  }
  private _errors: ValidationError[];

  constructor(errors: ValidationError[]) {
    this._errors = errors;
    this.errorsConverter(this._errors);
  }

  errorsConverter(errors: ValidationError[]): void {
    errors.forEach((error) => {
      if (!error.children.length) {
        this._convertedErrors.push({ property: error.property, constraints: error.constraints });
      } else {
        this.errorsConverter(error.children);
      }
    });
  }
}
