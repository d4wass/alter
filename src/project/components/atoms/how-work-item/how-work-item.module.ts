import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HowWorkItemComponent } from './how-work-item.component';

@NgModule({
  declarations: [HowWorkItemComponent],
  imports: [CommonModule],
  exports: [HowWorkItemComponent]
})
export class HowWorkItemModule {}
