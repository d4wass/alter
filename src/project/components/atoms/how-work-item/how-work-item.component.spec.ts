import { createComponentFactory, Spectator } from '@ngneat/spectator';
import { dataTest } from '../../../../utils/data-test';
import { HowWorkItemComponent } from './how-work-item.component';

describe('HowWorkItemComponent', () => {
  let component: HowWorkItemComponent;
  let spectator: Spectator<HowWorkItemComponent>;

  const createComponent = createComponentFactory({
    component: HowWorkItemComponent
  });

  const selectors = {
    indicator: () => spectator.query(dataTest('indicator'))
  };

  beforeEach(() => {
    spectator = createComponent();
    component = spectator.component;
  });

  it('should match to component', () => {
    expect(spectator.fixture).toMatchSnapshot();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  describe('@Inputs', () => {
    beforeEach(() => {
      component.itemContent = 'itemContent';
      component.itemNumber = '1';
      component.itemTitle = 'itemTitle';
    });

    it('should set [itemContent]', () => {
      const result = 'itemContent';
      expect(component.itemContent).toEqual(result);
    });
    it('should set [itemNumber]', () => {
      const result = '1';
      expect(component.itemNumber).toEqual(result);
    });
    it('should set [itemTitle]', () => {
      const result = 'itemTitle';
      expect(component.itemTitle).toEqual(result);
    });
  });

  describe('Component template', () => {
    beforeEach(() => {
      component.itemContent = 'itemContent';
      component.itemTitle = 'itemTitle';
    });
    it('should not display indicator when [itemNumber] is not defined', () => {
      spectator.detectComponentChanges();
      expect(component.itemNumber).toBeUndefined();
      expect(selectors.indicator()).toBeNull();
    });
    it('should display indicator when [itemNumber] is defined', () => {
      const result = '2';

      component.itemNumber = '2';
      spectator.detectComponentChanges();

      expect(selectors.indicator()?.innerHTML).toEqual(result);
    });
  });
});
