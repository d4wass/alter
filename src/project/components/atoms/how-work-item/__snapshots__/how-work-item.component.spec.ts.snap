// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowWorkItemComponent should match to component 1`] = `
<app-how-work-item>
  <div
    class="wrapper"
  >
    <div
      class="indicator"
    >
    </div>
    <div
      class="content"
    >
      <h2>
      </h2>
      <p>
      </p>
    </div>
  </div>
</app-how-work-item>
`;
